### DISCUSSION TITLE
OpenTelemetry capture request/response body?

### DISCUSSION THREAD

**Discussion by edeandrea** [quarkusio/quarkus#31663] [Q&A]
Is there a way in an OpenTelemetry trace to capture the request/response body of an HTTP call? It seems that other things are already being captured on both the client & server side of a call. Is there a way I could capture the json of the response?

![image](https://user-images.githubusercontent.com/363447/223532658-23811f2c-ea9e-48ac-879a-5ed39fae4f74.png)

[a comment by an automated account is not shown]

**Comment by radcortez** [comment]
Not currently. The request and response bodies are not part of the semantic conventions for capturing HTTP information: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md.

We discussed some time ago, about allowing the user to provide extractors to customize the trace information. Right now, the only way is to manually inject the `Span` and add the new attributes directly.

**Reply by edeandrea** [reply]
If I wanted to do that I assume I'd have to do it in some kind of filter, both on the jaxrs client (for outgoing request/responses) and on the resteasy reactive rest layer (for incoming request/responses)?

**Reply by radcortez** [reply]
Correct. That is how we implement the OTel stuff for the REST Client.

**Reply by edeandrea** [reply]
Can you point me to where those filters are in the otel stuff so I can take a look?

**Reply by brunobat** [reply]
Hi @edeandrea. The Classic rest client instrumentation wiring can be found here: https://github.com/quarkusio/quarkus/blob/main/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/intrumentation/restclient/OpenTelemetryClientFilter.java#L64
If you move a package up you will find the other instrumentations.

**Reply by brunobat** [reply]
BTW, I don't think it is a good practice to store requests and responses from prod environments for several reasons, being the main one PII.

**Reply by edeandrea** [reply]
> BTW, I don't think it is a good practice to store requests and responses from prod environments for several reasons, being the main one PII. 

Completely agree. My use case is for a demo where I was to visualize the requests and responses through a bunch of services as part of an overall user interaction.

Rather than build something myself to do that I thought piggy-backing on tools that already exist (Otel/Jaeger/etc) might be my fastest/easiest approach.

**Reply by brunobat** [reply]
Sure. That should work fine
